components/style/sharing/mod.rs

```diff
@@ -45,10 +129,35 @@ pub struct StyleSharingCandidate<E: TElement> {
     /// The element. We use SendElement here so that the cache may live in
     /// ScopedTLS.
     element: SendElement<E>,
-    /// The cached class names.
-    class_attributes: Option<Vec<Atom>>,
-    /// The cached result of matching this entry against the revalidation selectors.
-    revalidation_match_results: Option<BitVec>,
+    /// The cached data we store to avoid recomputing too much stuff.
+    cache: CachedStyleSharingData,
```

Comment:
Along with the rename of the struct suggested elsewhere, we should call this (and other names) |criteria| or |sharing_criteria|.